### DISCUSSION TITLE
Mitsuba variant "scalar_spectral_polarized" not found

### DISCUSSION THREAD

**Discussion by gioarma** [mitsuba-renderer/mitsuba3#969] [Q&A] [answered]
I'm following the basic example for polarized rendering. I compiled the system following the instructions, and modified the `mitsuba.conf` file in this way:
```
"enabled": [
        "scalar_rgb", "scalar_spectral", "cuda_ad_rgb", "llvm_ad_rgb", "scalar_spectral_polarized", "scalar_rgb_polarized"
    ]
```

In my notebook I first run 

```python
import mitsuba as mi
mi.set_variant('scalar_spectral_polarized')
```
and I get no error, so I guess the system is properly recognizing the poloarized variant.

However, when I import the `cbox_pol.xml` scene I get the error reported below. I did not modify in any way the .xml file, it is exactly the one provided on the website.
```python
scene = mi.load_file('Scenes/cbox_pol.xml')
```
```python
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
File C:\Program Files\mitsuba3\build\Release\python\mitsuba\__init__.py:107, in MitsubaVariantModule.__getattribute__(self, key)
    104 with _dr.scoped_rtld_deepbind():
    105     modules = (
    106         _import('mitsuba.mitsuba_ext'),
--> 107         _import('mitsuba.mitsuba_' + variant + '_ext'),
    108     )
    110 super().__setattr__('_modules', modules)

File C:\ProgramData\Miniconda3\envs\mitsuba_rendering\Lib\importlib\__init__.py:126, in import_module(name, package)
    125         level += 1
--> 126 return _bootstrap._gcd_import(name[level:], package, level)

File <frozen importlib._bootstrap>:1204, in _gcd_import(name, package, level)

File <frozen importlib._bootstrap>:1176, in _find_and_load(name, import_)

File <frozen importlib._bootstrap>:1140, in _find_and_load_unlocked(name, import_)

ModuleNotFoundError: No module named 'mitsuba.mitsuba_scalar_spectral_polarized_ext'

During handling of the above exception, another exception occurred:

AttributeError                            Traceback (most recent call last)
Cell In[12], line 1
----> 1 scene = mi.load_file('Scenes/cbox_pol.xml')

File C:\Program Files\mitsuba3\build\Release\python\mitsuba\__init__.py:253, in MitsubaModule.__getattribute__(self, key)
    251     sub_suffix = '' if submodule is None else f'.{submodule}'
    252     module = _sys.modules[f'mitsuba.{variant}{sub_suffix}']
--> 253     result = module.__getattribute__(key)
    255 # Add set_variant(), variant() and variant modules to the __dict__
    256 if submodule is None and key == '__dict__':

File C:\Program Files\mitsuba3\build\Release\python\mitsuba\__init__.py:113, in MitsubaVariantModule.__getattribute__(self, key)
    111 except ImportError as e:
    112     if str(e).startswith('No module named'):
--> 113         raise AttributeError('Mitsuba variant "%s" not found.' % variant)
    114     else:
    115         raise AttributeError(e)

AttributeError: Mitsuba variant "scalar_spectral_polarized" not found.
```

**Comment by wjakob** [comment]
You need to compile this variant yourself -- it's not one that we ship by default. Please read the documentation on how to make builds with additional variants of the system.

**Reply by gioarma** [reply]
Hello, thanks for your answer
I already followed all the instruction at the [Compiling the System](https://mitsuba.readthedocs.io/en/latest/src/developer_guide/compiling.html) page, and I'm getting the error reported above.
I looked for the documentation for building specific variants, but I did not find it.
I'm sorry but I'm not familiar with building with cmake

Any help is appreciated, thanks!

**Comment by merlinND** [comment]
If your custom Mitsuba build succeeded, then the most likely issue is that `import mitsuba as mi` picks up another version on your `PATH` (e.g. one installed via `pip`).
You can print `mi.variants()` and `mi.__file__` to check whether they match your custom build.

**Comment by gioarma** [comment, marked as answer]
Thank you, I finally understood what I needed to do. My understanding was that all variants were installed by cmake, and I just needed to specify the ones I wanted to use in the `mitsuba.conf`. 
Instead, cmake builds only the variants specified in the `mitsuba.conf` file. So, if one wants to add a new variant they need to change the `mitsuba.conf` file adding the desired variant, and then run again cmake to build again the whole thing.

**Reply by njroussel** [reply]
:+1: [This is the relevant part of the documentation.](https://mitsuba.readthedocs.io/en/latest/src/developer_guide/compiling.html#configuring-mitsuba-conf)
